### PR TITLE
add support for cloud based syncing (e.g. Google Drive) via rclone

### PR DESCRIPTION
**Pull request by rtanaka**

### CODE REVIEW

**Comment by cimryan** [comment]
Nice work!

Does this move files off of the drive or mirror them?

**Comment by rtanaka** [comment]
it's set to move... easy to change the behavior by editting the file `archive-clips.sh` and changing the rclone command from `move` to  `sync` or `copy`

```rclone move "$file_name" "$drive:$path" >> "$LOG_FILE" 2>&1 || echo ""```

i noticed the original version was just moving the files over so i did the same thing.

everything seems to be working okay using test data - i'll get a real run on my drive home from work tonight.

File: setup/pi/setup-teslausb [hunk 1]
```diff
@@ -210,6 +222,10 @@ if [ "$RSYNC_ENABLE" = true ]
 then
   get_script /root/bin verify-archive-configuration.sh run/rsync_archive
   get_script /root/bin configure-archive.sh run/rsync_archive
+elif [ "$RCLONE_ENABLE" = true ]
+then
+  get_script /root/bin verify-archive-configuration.sh run/clone_archive
```
Comment (cimryan) [inline setup/pi/setup-teslausb]:
```suggestion
  get_script /root/bin verify-archive-configuration.sh run/rclone_archive
```

**Comment by cimryan** [comment]
Please test this process if you haven't, already; I'm not sure you how got past line 227 in setup-teslausb .

**Comment by rtanaka** [comment]
@cimryan - sorry about that. i'll do a fresh install with a new sdcard and confirm that everything is working.

**Comment by rtanaka** [comment]
okay - i started with a fresh sd card and confirmed the install works. i ran into two issues that didn't come up during testing from a clean install.

1. rclone is complaining about HOME not being set so it can't find the directory. i can get around this one by passing in the config file from when calling the script.

2. it looks like rclone updates it's token from time to time but as you know the file system is read-only.

suggestions on how to deal with item 2?

**Comment by cimryan** [comment]
Do we get to control where the token gets written? We could add another partition after the backingfiles partition and leave that one read/write.

**Comment by rtanaka** [comment]
yes - we can specify the config location upon creation of the config. 

i like the idea of having a read/write partition. we can store other configs in there too. also things like fake-hwclock (which will fix the time sync issue) needs a writeable directory. thoughts on size -- 5MB should be more than enough?

**Comment by cimryan** [comment]
The instructions currently require a MicroSD card which is at least 8GB in size; I think we can reserve up to 100MB, without issue.

**Comment by rtanaka** [comment]
Are you okay with accepting this PR first? It's is marked as experimental anyway... I can then submit another one with the changes to support the new partition hopefully over the weekend.

**Comment by cimryan** [comment]
I've merged into the branch u/rtanaka/rclone_archive . I'm not inclined to merge to master with known bugs, but if there's a compelling reason to do so I could be convinced.

**Pull request merged**